### PR TITLE
feat: use peer-pad-core with js-ipfs 0.30

### PR DESCRIPTION
**Pull request by diasdavid**
Let's give this a spin

Depends on:

- [x] https://github.com/ipfs-shipyard/peer-pad-core/pull/21

### CODE REVIEW

**Comment by jacobheun** [comment]
@diasdavid we'll need to add the package-lock back in. We've had stability issues with peer-pad in prod due to moving package versions so we switched to using package locks to improve stability.

**Comment by diasdavid** [comment]
Getting JSON parse errors

 in Firefox 

image

in Chrome

image

I believe this is connected to the fact that the PubSub API changed. Need to debug at the y-ipfs-connector level

**Comment by diasdavid** [comment]
image

Seems to be working now. Can someone else test it? Thank you!

**Comment by diasdavid** [comment]
I'm not sure if this is ok, but current PeerPad makes new peerpad break:

image

**Comment by gpestana** [comment]
It works as expected on my machine. Tested with Chrome 67.0.x and Brave 67.0.x 👌

**Comment by jacobheun** [comment]
Chrome and Firefox are communicating okay for me. There is some bugginess with the tabs, but that looks to be an existing issue.

@diasdavid peer-pad is setup to do a staging deploy for merges to master. I talked to @VictorBjelkholm last week about needing to get the DNS entry setup, https://github.com/ipfs-shipyard/peer-pad/issues/152#issuecomment-396248060. If we get that resolved we can test this in staging a bit more together, and then cut a production release via the `production` branch

**Comment by diasdavid** [comment]
NEW <-> NEW works fine

image

It's NEW <-> Current that fails. Seems that something changed on the data model (previously was not JSON??)

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "jest": "20.0.4",
     "milliseconds": "^1.0.3",
     "object-assign": "4.1.1",
-    "peer-pad-core": "^0.12.0",
+    "peer-pad-core": "ipfs-shipyard/peer-pad-core#js-ipfs0.30",
```
Comment (diasdavid) [inline package.json]: - [ ] Change this before merge

**Comment by gpestana** [comment]
NEW <-> old failures should not matter much as soon as you deploy the new version. I'm not sure what happens to those using the older version, but I reckon there might not be many peerpad users atm 🤔

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+
+matrix:
+  include:
+  - node_js: stable
+    env: CXX=g++-4.8
+
 script:
-  - npm run lint && npm test && npm run build && npm run test:e2e:ci
+  - npm run lint
+  - npm test
```
Comment (jacobheun) [inline .travis.yml]: `npm run build` needs to be run prior to the e2e tests so we have the static build to tests against.

**Comment by VictorBjelkholm** [comment]
@gpestana 
> NEW <-> old failures should not matter much as soon as you deploy the new version

While it's true currently that PeerPad doesn't have many current users, we should be in the mindset that versions should be compatible in-between upgrades, for many reasons. Most importantly, users should be able to pull down a version and use it locally and still be compatible with other clients. If exact version is enforced, this becomes very brittle. Secondly, during the updating of dns records, different users will get different dns replies, and the application will be broken.

But, not a huge issue right now, but we need to care about this in the future.

**Comment by gpestana** [comment]
@VictorBjelkholm 

> While it's true currently that PeerPad doesn't have many current users, we should be in the mindset that versions should be compatible in-between upgrades

Completely agree 👍

**Comment by diasdavid** [comment]
@VictorBjelkholm you are preaching the mess to the quire here :) I 💯 agree with your statement and I see how my comment got taken out of context.

Peerpad is in a inconsistent state for 7 months, so much that things are not working, out of date or there were migrations on the data model that made things break. This was caused by lack of resources + project shuffling + lack of consistent product management. 

@pgte is now focused to level everything to a new level with peer-crdt and peer-star-app, a new version will come out very soon. Meanwhile, this update comes to make things slightly better and move the needed a bit forward on the "maintenance side".

**Comment by diasdavid** [comment]
<a href="https://www.youtube.com/watch?v=g0l738fq3C0">image</a>
Url: https://www.youtube.com/watch?v=g0l738fq3C0

File: .travis.yml (hunk 2 above)
Comment (diasdavid) [reply .travis.yml]: fixed. thanks!

Comment (VictorBjelkholm) [reply .travis.yml]: Why is this file not removed instead?

**Pull request merged**